dali/base/dasds.cpp

```diff
@@ -270,18 +262,28 @@ class CLCLockBlock : public CInterface
     {
         lock.unlock();
     }
-    void changeToWrite(unsigned timeout)
-    {
-        /* NB: this intentionally clears the lock, before re-establishing a write lock
-        * because we do not want to keep this read lock whilst waiting for a write,
-        * as other write transactions would pile up.
-        */
-        lock.unlock();
-        lock.lockWrite(timeout);
-    }
+    void changeToWrite(unsigned timeout);
```

Comment:
Should this have same prototype as above?

Reply:
yes, it's in conditional code which isn't on unless a #define is changed, but should be changed.